Backend/lib/message-service/src/Lib/Domain/Action/UI/Message.hs

```diff
@@ -63,10 +62,10 @@ instance ToJSON MessageAPIEntityResponse where
 newtype MessageReplyReq = MessageReplyReq {reply :: Text}
   deriving (Generic, ToSchema, ToJSON, FromJSON)
 
-messageList :: (Id SP.Person, Id DM.Merchant) -> Maybe Int -> Maybe Int -> Flow [MessageAPIEntityResponse]
-messageList (driverId, _) mbLimit mbOffset = do
-  person <- Esq.runInReplica (QP.findById driverId >>= fromMaybeM (PersonNotFound driverId.getId))
-  messageDetails <- Esq.runInReplica $ MRQ.findByDriverIdAndLanguage (cast driverId) (fromMaybe ENGLISH person.language) mbLimit mbOffset
+messageList :: (HasField "language" a (Maybe Language), MonadFlow m, EsqDBReplicaFlow m r, EsqDBFlow m r) => (Id Domain.Person, Id Domain.Merchant) -> Maybe Int -> Maybe Int -> m (Maybe a) -> m [MessageAPIEntityResponse]
+messageList (personId, _) mbLimit mbOffset findByPersonId = do
+  person <- findByPersonId >>= fromMaybeM (PersonNotFound personId.getId)
```

Comment:
you can pass only the required things from person instead of passing the findById function

Reply:
Same.